Repro here:

https://github.com/fonttools/fonttools/actions/runs/25217674832/job/73941813830?pr=4085#step:5:4125

```
>       assert result == pytest.approx(expected)
E       assert 818.1818181818181 == 809.0909090909091 ± 8.1e-04
E         
E         comparison failed
E         Obtained: 818.1818181818181
E         Expected: 809.0909090909091 ± 8.1e-04
```

--- a/Tests/designspaceLib/designspace_test.py
+++ b/Tests/designspaceLib/designspace_test.py
@@ -1232,3 +1232,89 @@ def test_map_backward_many_to_one():
     result = ax.map_backward(800)
     expected = 800 + (800 - 780) / (1000 - 780) * (900 - 800)  # ~809.09
     assert result == pytest.approx(expected)
+
+
+def test_map_backward_many_to_one_mid_range():
+    """Test map_backward with a flat segment in the middle of the axis range."""
+    ax = AxisDescriptor()
+    ax.name = "Weight"
+    ax.tag = "wght"
+    ax.minimum = 100
+    ax.default = 100
+    ax.maximum = 1000
+    # Many-to-one in the middle: both user=400 and user=500 map to design=50
+    ax.map = [
+        (100, 10),
+        (400, 50),
+        (500, 50),
+        (700, 80),
+        (900, 100),
+    ]
+
+    # Before the flat segment: interpolate between (10, 100) and (50, 400)
+    result = ax.map_backward(45)
+    expected = 100 + (45 - 10) / (50 - 10) * (400 - 100)  # 362.5
+    assert result == pytest.approx(expected)
+
+    # After the flat segment: interpolate between (50, 500) and (80, 700)
+    result = ax.map_backward(65)
+    expected = 500 + (65 - 50) / (80 - 50) * (700 - 500)  # 600
+    assert result == pytest.approx(expected)
+
+
+def test_map_backward_many_to_one_at_flat_value():
+    """Test map_backward queried exactly at the flat design value."""
+    ax = AxisDescriptor()
+    ax.name = "Weight"
+    ax.tag = "wght"
+    ax.minimum = 100
+    ax.default = 100
+    ax.maximum = 1000
+    ax.map = [
+        (100, 100),
+        (800, 780),
+        (900, 1000),
+        (1000, 1000),
+    ]
+
+    # map_backward(1000) should return 900 (first user value reaching design=1000)
+    assert ax.map_backward(1000) == 900
+
+    # map_backward(780) should still return 800
+    assert ax.map_backward(780) == 800
+
+
+def test_map_backward_many_to_one_unsorted_input():
+    """Test map_backward with map entries not in user-value order."""
+    ax = AxisDescriptor()
+    ax.name = "Weight"
+    ax.tag = "wght"
+    ax.minimum = 100
+    ax.default = 100
+    ax.maximum = 1000
+    # Same map as the extreme case, but entries shuffled
+    ax.map = [
+        (1000, 1000),
+        (100, 100),
+        (900, 1000),
+        (800, 780),
+    ]
+
+    result = ax.map_backward(800)
+    expected = 800 + (800 - 780) / (1000 - 780) * (900 - 800)  # ~809.09
+    assert result == pytest.approx(expected)
+
+
+def test_map_backward_single_entry():
+    """Test map_backward with a single map entry."""
+    ax = AxisDescriptor()
+    ax.name = "Weight"
+    ax.tag = "wght"
+    ax.minimum = 100
+    ax.default = 100
+    ax.maximum = 1000
+    ax.map = [(100, 200)]
+
+    assert ax.map_backward(200) == 100
+    assert ax.map_backward(300) == 200
+    assert ax.map_backward(100) == 0

